my_api/search_sync.py

```diff
@@ -3,9 +3,13 @@
 
 Thank you to: https://github.com/tiangolo/fastapi/issues/2713
 """
+
+from __future__ import annotations
```

Comment:
Without this the style of types you are using will cause error in Python <3.10. I added it since the pyproject.toml says 3.8+ are supported.